docs/security/gpg-key-for-ssh-authentication.md

```diff
@@ -0,0 +1,477 @@
+    Current allowed actions: Sign Encrypt 
+    
+       (S) Toggle the sign capability
+       (E) Toggle the encrypt capability
+       (A) Toggle the authenticate capability
+       (Q) Finished
+    
+    Your selection? S
+                     
+    Possible actions for a RSA key: Sign Encrypt Authenticate 
+    Current allowed actions: Encrypt 
+    
+       (S) Toggle the sign capability
+       (E) Toggle the encrypt capability
+       (A) Toggle the authenticate capability
+       (Q) Finished
+    
+    Your selection? E
+                     
+    Possible actions for a RSA key: Sign Encrypt Authenticate 
+    Current allowed actions: 
+    
+       (S) Toggle the sign capability
+       (E) Toggle the encrypt capability
+       (A) Toggle the authenticate capability
+       (Q) Finished
+    
+    Your selection? A
+                     
+    Possible actions for a RSA key: Sign Encrypt Authenticate 
+    Current allowed actions: Authenticate 
+    
+       (S) Toggle the sign capability
+       (E) Toggle the encrypt capability
+       (A) Toggle the authenticate capability
+       (Q) Finished
+    
+    Your selection? Q
+    RSA keys may be between 1024 and 4096 bits long.
+    What keysize do you want? (2048) 4096
+    Requested keysize is 4096 bits       
+    Please specify how long the key should be valid.
+             0 = key does not expire
+          <n>  = key expires in n days
+          <n>w = key expires in n weeks
+          <n>m = key expires in n months
+          <n>y = key expires in n years
+    Key is valid for? (0) 1y
+    Key expires at Sun Apr  9 18:49:58 2017 AEST
+    Is this correct? (y/N) y
+    Really create? (y/N) y  
+    We need to generate a lot of random bytes. It is a good idea to perform some other action (type on the keyboard, move the mouse, utilize the disks) during the prime generation; this gives the random number generator a better chance to gain enough entropy.
+    
+    pub  4096R/71735D23  created: 2016-04-09  expires: 2017-04-09  usage: SC  
+                         trust: ultimate      validity: ultimate
+    sub  4096R/693C5635  created: 2016-04-09  expires: 2017-04-09  usage: E   
+    sub  4096R/48B9C23C  created: 2016-04-09  expires: 2017-04-09  usage: A   
+    [ultimate] (1). Your Name (Test for GPG -> SSH) <you@yoursite.net>
+
+### Securing Your GPG Key
+
+{: .caution}
+>
+> If you fail to back up or otherwise secure your key, any hardware failure will lead to you being unable to access your Linode with this key. If you lock out password access through SSH, this may be disastrous.
+
+You should always have a backup of your private key in case something goes wrong and you end up locked out of everything. This private key, along with the instructions in this guide, will be enough to get your setup working again if you need to start afresh on a new computer.
+
+1. Backup your `~/.gnupg` folder by typing:
+
+        mv ~/.gnupg/ /Volumes/USB_DEVICE/.gnupg/
```

Comment:
Thank you for noticing that—I did mean `cp`.

Most of the folder is needed for the backup, so I think a convoluted command or shell script probably strays a little too far from a simplistic one-liner. We have two options for copying the socket file if we consider that.

1. Use `cp -r` and tell the user to ignore the errors
2. Use `rsync -r --exclude=S.gpg-agent ~/.gnupg /Volumes/USB_DEVICE`, which can exclude files matching that pattern.

In my changes, I've gone with the first option, since I'm not sure if `rsync` is standard and the output messages are just warnings.